src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,717 @@
+      const auto &ev = **it;
+      std::string key = "KEY" + NumToString(keyCount);
+      std::string value = "VALUE" + NumToString(keyCount);
+      std::string name = Name(ev);
+      LowerCase(&name);
+      code_.SetValue(key, name);
+      code_.SetValue(value, enum_def.ToString(ev));
+      enumCode += "{{" + key + "}} = {{" + value + "}}, ";
+      keyCount++;
+    }
+    code_ += enumCode.substr(0, enumCode.size() - 2);
+    code_ += "}\n";
+  }
+
+  void GenLookup(const FieldDef &key_field) {
+    code_.SetValue("OFFSET", NumToString(key_field.value.offset));
+    std::string offsetReader =
+        "Table.offset(Int32(fbb.capacity) - tableOffset, vOffset: {{OFFSET}}, "
+        "fbb: fbb)";
+    std::string spacing = "\t\t";
+    std::string doubleSpacing = spacing + "\t";
+
+    code_.SetValue("TYPE", GenType(key_field.value.type));
+    code_ +=
+        "\tpublic static func lookupByKey(vector: Int32, key: {{TYPE}}, fbb: "
+        "FlatBuffer) -> {{VALUENAME}}? {";
+
+    if (key_field.value.type.base_type == BASE_TYPE_STRING)
+      code_ += spacing + "let key = key.utf8.map { $0 }";
+    code_ += spacing +
+             "var span = fbb.read(def: Int32.self, position: Int(vector - 4))";
+    code_ += spacing + "var start: Int32 = 0";
+    code_ += spacing + "while span != 0 {";
+    code_ += doubleSpacing + "var middle = span / 2";
+    code_ +=
+        doubleSpacing +
+        "let tableOffset = Table.indirect(vector + 4 * (start + middle), fbb)";
+    if (key_field.value.type.base_type == BASE_TYPE_STRING) {
+      code_ += doubleSpacing + "let comp = Table.compare(" + offsetReader +
+               ", key, fbb: fbb)";
+    } else {
+      code_ += doubleSpacing +
+               "let comp = fbb.read(def: {{TYPE}}.self, position: Int(" +
+               offsetReader + "))";
+    }
+
+    code_ += doubleSpacing + "if comp > 0 {";
+    code_ += doubleSpacing + "\tspan = middle";
+    code_ += doubleSpacing + "} else if comp < 0 {";
+    code_ += doubleSpacing + "\tmiddle += 1";
+    code_ += doubleSpacing + "\tstart += middle";
+    code_ += doubleSpacing + "\tspan -= middle";
+    code_ += doubleSpacing + "} else {";
+    code_ += doubleSpacing + "\treturn {{VALUENAME}}(fbb, o: tableOffset)";
+    code_ += doubleSpacing + "}";
+    code_ += spacing + "}";
+    code_ += spacing + "return nil";
+    code_ += "\t}";
+  }
+
+  std::string GenOffset() { return "let o = {{ACCESS}}.offset({{OFFSET}}); "; }
+
+  std::string GenReaderMainBody(std::string optional = "") {
+    return "\tpublic var {{VALUENAME}}: {{VALUETYPE}}" + optional + " { ";
+  }
+  std::string GenReader(std::string type, std::string at = "{{OFFSET}}") {
+    return "{{ACCESS}}.readBuffer(of: {{" + type + "}}.self, at: " + at + ")";
+  }
+
+  std::string GenArrayMainBody(std::string optional = "") {
+    return "\tpublic func {{VALUENAME}}(at index: Int32) -> {{VALUETYPE}}" +
+           optional + " { ";
+  }
+
+  std::string GenIndirect(std::string reading) {
```

Comment:
Please inline all functions called just once.

Reply:
Do you mean remove them outside the class? and keep them in the swift namespace?